src/bit_array.cr

```diff
@@ -165,6 +165,78 @@ struct BitArray
     end
   end
 
+  # :inherit:
+  def all? : Bool
+    bit_index, sub_index = @size.divmod(32)
+
+    bit_index.times do |i|
+      return false unless @bits[i] == UInt32::MAX
+    end
+
+    return true if sub_index == 0
```

Comment:
FWIW this line is not necessary. Up to you.
```suggestion
```

Reply:
`@bits[bit_index]` would be out of bounds without this line if `@size` is a multiple of 32.

Reply:
Then there's no spec for that ;)
Because I removed the line and all specs passed

Reply:
That's because in that case `x & 0 == 0` would always be true, and any failure would be a segmentation fault instead; to test for such a failure we have to guarantee that `Pointer.malloc` never over-allocates, since `@bits` is a `Pointer`, not a `Slice`. That is currently beyond the scope of those specs.